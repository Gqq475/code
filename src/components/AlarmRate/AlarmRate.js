import React from 'react'
import classes from './AlarmRate.scss'
import { connect } from 'react-redux'

type Props = {
  token: String
};

export class AlarmRate extends React.Component {
  props: Props;

  constructor (props) {
    super(props)
    this.state = {
      url: ''
    }
  }

  componentDidMount () {
    let url = 'https://bi-proxy.cloud.k2data.com.cn/1.0.0/workbooks/' +
      '690abb9f-9700-403e-8f70-3ce37573d453/views/a647abb3-793a-48e9-ba9d-9a7961291420'

    fetch(url, {
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': this.props.token
      }
    })
    .then(response => response.json())
    .then(data => {
      this.setState({ url: data.result.contentUrl })
    })
    .catch(err => console.log(err))
  }

  render () {
    return (
      <div className={classes['AlarmRate-container']}>
        {
          <iframe className={classes.iframe} src={this.state.url} />
        }
      </div>
    )
  }
}

export default connect(
  (state) => ({
    token: state.auth.token
  }),
  (dispatch, state) => {
    return {
      dispatch
    }
  }
)(AlarmRate)
