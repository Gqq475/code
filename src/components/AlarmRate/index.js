import AlarmRate from './AlarmRate'
export default AlarmRate
