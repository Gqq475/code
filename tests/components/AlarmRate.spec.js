import React from 'react'
import { AlarmRate } from 'components/AlarmRate/AlarmRate'

describe('(Component) AlarmRate', () => {
  it('render normally', () => {
    const wrapper = shallow(
      <AlarmRate />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
