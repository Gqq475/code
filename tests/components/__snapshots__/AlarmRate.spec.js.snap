// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) AlarmRate render normally 1`] = `
<div
  className="AlarmRate-container"
>
  <iframe
    className="iframe"
    src=""
  />
</div>
`;
